content/blog/k8s-operator/index.md

```diff
@@ -0,0 +1,98 @@
+vizier-cloud-connector-854f8bb487-d69kk   1/1     Running   0          2d20h
+vizier-metadata-79f8764589-hmz59          1/1     Running   0          2d20h
+vizier-pem-crg62                          1/1     Running   12         2d20h
+vizier-pem-r2xsn                          1/1     Running   4          2d20h
+vizier-proxy-f584dc9c8-4gb72              1/1     Running   0          2d20h
+vizier-query-broker-ddbc89b-wftbz         1/1     Running   0          2d20h
+```
+
+After the addition of the CRD, the entire state of the application can be summarized with one `kubectl
+describe vizier` command:
+
+```bash
+kubectl describe vizier
+
+Status:
+ Last Reconciliation Phase Time:  2021-11-05T22:30:56Z
+ Reconciliation Phase:            Ready
+ Version:                         0.9.11
+ Vizier Phase:                    Healthy
+```
+
+### 🟢 Cleanliness and consistency
+
+Configuration options live in one place (the CRD) rather than spread out across many configmaps. The values in the CRD can be viewed as the source of truth, and is the single place where users need to make their modifications when adjusting the config.
+
+Pixie originally had four configmaps:
+
+```bash
+pl-cloud-config
+pl-cloud-connector-tls-config
+pl-cluster-config
+pl-tls-config
+```
+
+These configMaps are now represented by a [single CRD](https://github.com/pixie-io/pixie/blob/main/k8s/operator/crd/base/px.dev_viziers.yaml).
+
+### 🟢 Auto-recovery
+
+The operator can monitor the overall state of the application and apply whatever changes necessary to get the application into a healthy state. This is especially beneficial for persistent systems, or applications that need to be highly available.
+
+NATS is a major dependency of Pixie, and enables most of Pixie’s pod-to-pod communication. Occasionally we have seen the NATS instance fail, and require a redeploy to recover. The operator can monitor NATS’s status and redeploy when necessary.
+
+### 🔴 Loss of user control
+
+The operator is responsible for deploying K8s resources that are abstracted away from the user. However, many users prefer to know exactly what is deployed on their system. Since it is the operator’s responsibility to manage these resources, the operator may also unknowingly overwrite any manual user changes.
+
+In Pixie’s case, the user deploys only the [operator’s YAMLs](https://github.com/pixie-io/pixie/tree/main/k8s/operator) to get started. Pixie’s operator actually deploys [more resources](https://github.com/pixie-io/pixie/tree/main/k8s/vizier) to their cluster, which are not included in these initial YAMLs.
+
+### 🔴 Maintenance burden
+
+The operator is an additional piece of code that needs to be maintained and updated, alongside with the actual application itself. The more powerful the operator, the more complex its logic. The operator may be responsible for keeping the application up-to-date, but what happens when [the operator itself](https://olm.operatorframework.io/) needs to be updated?
```

Comment:
Maybe we should have the whole phrase "operator itself needs to be updated" link to https://olm.operatorframework.io/, rather than just "the operator itself".

Reply:
Since the link points to the operator (but not specific info about how to update the operator), I thought this was more clear. WDYT?

Reply:
It points to the life cycle manager, which is how you would update the operator.